src/Microsoft.Azure.SignalR.SignalRService/ServiceManagerBuilder.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.Azure.SignalR.SignalRService
+{
+    public class ServiceManagerBuilder
+    {
+        public ServiceManagerBuilder WithCredential(string conenctionString)
+        {
+            throw new NotImplementedException();
+        }
+
+        public ServiceManagerBuilder WithCredential(string endpoint, string accessToken)
+        {
+            throw new NotImplementedException();
+        }
+
+        public ServiceManagerBuilder WithCredential(string endpoint, IList<string> accessTokens)
+        {
+            throw new NotImplementedException();
+        }
+
+        public ServiceManagerBuilder WithOptions(Action<ServiceUtilityOptions> options)
+        {
+            throw new NotImplementedException();
+        }
+
+        public ServiceHubContext CreateHubContextAsync(string hubName)
+        {
+            throw new NotImplementedException();
+        }
+
+        public ServiceManager Build()
```

Comment:
Async?

Reply:
`Build` method may need to establish connection asynchronously to service (e.g. for monitoring the whole service), thus  `async` is more appropriate? Is it your concern?

_For now, we design establishing connection asynchronously to service in `ServiceHubContext CreateHubContextAsync(string hubName)`._

Reply:
Sounds good, as long as it's not designed as an async operation.

Reply:
@chenkennt 

I consider that `ServiceManager` is designed for used as a collection of tools which are managing service. 

`Build` is only used to set credentials and options for tools in service manager. 

If you want to use some tool, you need to create it first. ServiceManager.Create<Tool>(parameters...) is used to create tool. Tool itself control its ws connection (to service) lifetime and other resources.

Thus `Build` is not designed as an async operation.